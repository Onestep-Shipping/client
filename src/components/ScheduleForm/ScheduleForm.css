.css-10nd86i {
  width: 260px;
  outline: none;
  margin-right: 40px;
  border-radius: 4px;
}

.css-1aya2g8 {
  border: solid #14B9D7 2px !important;
}