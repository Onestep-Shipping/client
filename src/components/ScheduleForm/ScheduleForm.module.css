.formContainer {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: center;
  margin-top: 20px;
  width: 461px;
  height: 427px;
  border-radius: 10px;
  border: solid #474747 1px;
  padding: 20px;
}

.infoContainer {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: center;
}

.textfieldContainer {
  width: 100%;
  margin-bottom: 50px;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
}

.scheduleLabel {
  color: #474747;
  font-size: 23px;
  margin-bottom: 10px;
}

.toDate {
  width: 130px;
  height: 30px;
  padding: 20px;
  font-size: 16px;
  padding: 5px 15px;
  border-radius: 4px;
  border: none;
  outline: none;
  border: solid #474747 1px;
}

.fromDate {
  width: 130px;
  height: 30px;
  padding: 20px;
  font-size: 16px;
  padding: 5px 15px;
  border-radius: 4px;
  border: none;
  outline: none;
  border: solid #474747 1px;
}

.carrierSelector {
  width: 200px;
  height: 35px;
  padding: 20px;
  font-size: 16px;
  padding: 5px 15px;
  border-radius: 4px;
  border: none;
  outline: none;
  border: solid #474747 1px;
  background-color: white;
  box-shadow: none;
}

.findButton {
  width: 100%;
  padding: 10px 20px;
  background-color: #224DAA;
  border: none;
  outline: none;
  color: white;
  font-size: 20px;
  border-radius: 3px;
  cursor: pointer;
  margin-bottom: 15px;
}

.findButton:hover {
  background-color: rgb(20, 55, 131);
}