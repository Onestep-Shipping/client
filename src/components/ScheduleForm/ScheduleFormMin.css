.search-container2 {
  width: fit-content;
  height: fit-content;
  margin-bottom: 20px;
}

.textfield-container2 {
  margin-bottom: 20px;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
}

.css-1aya2g8 {
  outline: none !important;
  border: none !important;
  border-radius: 2px !important;
}

.css-10nd86i {
  width: 300px;
  outline: none;
  margin-right: 40px;
  border-radius: 4px;
  border: solid #14B9D7 2px;
}

.schedule-label-text2 {
  color: #14B9D7;
  font-size: 20px;
  margin-right: 20px;
}

#from-date2 {
  width: 130px;
  height: 25px;
  padding: 20px;
  font-size: 16px;
  padding: 5px 15px;
  border-radius: 4px;
  border: solid #14B9D7 2px;
  outline: none;
}

#to-date2 {
  width: 130px;
  height: 25px;
  padding: 20px;
  font-size: 16px;
  padding: 5px 15px;
  border-radius: 4px;
  border: solid #14B9D7 2px;
  outline: none;
}

#carrier-selector2 {
  width: 80%;
  height: 35px;
  padding: 20px;
  font-size: 16px;
  padding: 5px 15px;
  border-radius: 4px;
  border: solid #14B9D7 2px;
  outline: none;
}

#find-button2 {
  padding: 10px 15px;
  background-color: #BF037A;
  border: none;
  outline: none;
  color: white;
  font-size: 20px;
  border-radius: 3px;
  cursor: pointer;
}