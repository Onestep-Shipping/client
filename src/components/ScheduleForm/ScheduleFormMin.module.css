.formContainer {
  width: fit-content;
  height: fit-content;
  margin-bottom: 20px;
}

.infoContainer {
  margin-bottom: 20px;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
}

.textfieldContainer {
  width: 88%;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.scheduleLabel {
  color: #14B9D7;
  font-size: 20px;
  margin-right: 20px;
}

.toDate {
  width: 130px;
  height: 25px;
  padding: 20px;
  font-size: 16px;
  padding: 5px 15px;
  border-radius: 4px;
  border: solid #14B9D7 2px;
  outline: none;
}

.fromDate {
  width: 130px;
  height: 25px;
  padding: 20px;
  font-size: 16px;
  padding: 5px 15px;
  border-radius: 4px;
  border: solid #14B9D7 2px;
  outline: none;
}

.carrierSelector {
  width: 80%;
  height: 35px;
  padding: 20px;
  font-size: 16px;
  padding: 5px 15px;
  border-radius: 4px;
  border: solid #14B9D7 2px;
  outline: none;
}

.findButton {
  padding: 10px 15px;
  background-color: #BF037A;
  border: none;
  outline: none;
  color: white;
  font-size: 20px;
  border-radius: 3px;
  cursor: pointer;
}

.findButton:hover {
  background-color: rgb(163, 0, 103);
}