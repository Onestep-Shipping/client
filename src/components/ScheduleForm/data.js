const CARRIERS = ['Hapag-Lloyd', 'Maersk', 'YangMing', 'ONE', 'MSC', 'APL'];

const FROM_LOCATIONS = [
  {label: 'Vancouve (BC), CANADA', value: 'CAVAN'},
  {label: 'Toronto (ON), CANADA', value: 'CATOR'},
  {label: 'Montreal (QC), CANADA', value: 'CAMTR'},
  {label: 'Halifax (NS), CANADA', value: 'CAHAL'},

  {label: 'Los Angeles (CA), USA', value: 'USLAX'},
  {label: 'Oakland (CA), USA', value: 'USOAK'},
  {label: 'Long Beach (CA), USA', value: 'USLGB'},
  {label: 'Seattle (WA), USA', value: 'USSEA'},
  {label: 'Tacoma (CA), USA', value: 'USTIW'},
  {label: 'New York (NY), USA', value: 'USNYC'},
  {label: 'Savannah (GA), USA', value: 'USSAV'},
];

const TO_LOCATIONS = [
  {label: 'Keelung, CHINA', value: 'TWKEL'},
  {label: 'Busan, KOREA', value: 'KRPUS'},
  {label: 'Singapore, SINGAPORE', value: 'SGSIN'},
  {label: 'Port Klang, MALAYSIA', value: 'MYPKG'},
  {label: 'Jakarta, INDONESIA', value: 'IDJKT'},
  {label: 'Bangkok, THAILAND', value: 'THBKK'},
  {label: 'Ho Chi Minh City, VIETNAM', value: 'VNSGN'},
  {label: 'Taichung, TAIWAN', value: 'TWTXG'},
];

export {FROM_LOCATIONS, TO_LOCATIONS, CARRIERS};
