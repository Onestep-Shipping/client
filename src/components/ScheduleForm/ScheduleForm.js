import React, {useState} from 'react';
import {useHistory} from 'react-router-dom';
import Select from 'react-select';
import DatePicker from 'react-datepicker';
import {
  FROM_LOCATIONS, TO_LOCATIONS, CARRIERS,
} from '../../data/ScheduleFormData.js';

import './ScheduleForm.css';
import 'react-datepicker/dist/react-datepicker.css';

const ScheduleForm = () => {
  const history = useHistory();

  const handleSubmit = (e) => {
    e.preventDefault();
    if (fromLocation !== '' && toLocation !== '') {
      history.push('/schedule');
    }
  };

  const today = new Date();
  const [startDate, setStartDate] = useState(today);
  const [endDate, setEndDate] = useState(today);

  const [fromLocation, setFromLocation] = useState('');
  const [toLocation, setToLocation] = useState('');
  const [carrier, setCarrier] = useState(CARRIERS[0]);

  return (
    <form className="schedule-form-cotainer" onSubmit={handleSubmit}>
      <text className="schedule-label">From</text>
      <div className="textfield-container">
        <Select
          options={FROM_LOCATIONS} placeholder="Location"
          onChange={(opt) => setFromLocation(opt.label)} clearable/>
        <DatePicker
          id="from-date"
          selected={startDate}
          onChange={setStartDate}
          placeholderText="Select a day"
        />
      </div>
      <text className="schedule-label">To</text>
      <div className="textfield-container">
        <Select options={TO_LOCATIONS} placeholder="Location"
          onChange={(opt) => setToLocation(opt.label)} clearable/>
        <DatePicker
          id="to-date"
          selected={endDate}
          minDate={startDate}
          onChange={setEndDate}
        />
      </div>
      <text className="schedule-label">Carrier</text>
      <div className="textfield-container">
        <select
          id="carrier-selector"
          onChange={(carrier) => setCarrier(carrier)} >
          {
            CARRIERS.map((opt, ind) => {
              return (<option value={opt} key={ind}>{opt}</option>);
            })
          }
        </select>
      </div>
      <button id="find-button" type="submit">Find</button>
    </form>
  );
};

export default ScheduleForm;
