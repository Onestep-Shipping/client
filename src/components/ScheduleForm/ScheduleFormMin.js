import React, {useState} from 'react';
import './ScheduleFormMin.css';
import Select from 'react-select';
import DatePicker from 'react-datepicker';
import {FROM_LOCATIONS, TO_LOCATIONS, CARRIERS} from './data.js';

const ScheduleFormMin = () => {
  const today = new Date();
  const [startDate, setStartDate] = useState(today);
  const [endDate, setEndDate] = useState(today);

  return (
    <div class="search-container2">
      <div class="textfield-container2">
        <text class="schedule-label-text2">From</text>
        <Select options={FROM_LOCATIONS} placeholder="Location"/>
        <DatePicker
          id="from-date2"
          selected={startDate}
          onChange={setStartDate}
          placeholderText="Select a day"
        />
      </div>
      <div class="textfield-container2">
        <text class="schedule-label-text2">  </text>
        <text class="schedule-label-text2">To</text>
        <Select options={TO_LOCATIONS} placeholder="Location"/>
        <DatePicker
          id="to-date2"
          selected={endDate}
          onChange={setEndDate}
          minDate={startDate}
          placeholderText="Select a day"
        />
      </div>
      <div class="textfield-container2">
        <text class="schedule-label-text2">Carrier</text>
        <select id="carrier-selector2">
          {
            CARRIERS.map((opt) => {
              return (<option value={opt}>{opt}</option>);
            })
          }
        </select>
      </div>
      <button id="find-button2">Find</button>
    </div>
  );
};

export default ScheduleFormMin;
