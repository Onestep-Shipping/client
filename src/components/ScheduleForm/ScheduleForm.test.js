import React from 'react';
import ScheduleForm from './ScheduleForm';
import ScheduleFormProvider from '../../context/ScheduleFormContext.js';
import { shallow } from 'enzyme';
import styles from './ScheduleForm.module.css';

const mockSubmit = jest.fn();

let wrapper = shallow( 
  <ScheduleFormProvider>
    <ScheduleForm styles={styles} onSubmit={mockSubmit}/>
  </ScheduleFormProvider>
);

describe('ScheduleForm', () => {
  it('should render the ScheduleForm Component correctly', () => {   
    expect(ScheduleForm).toMatchSnapshot();
  });
  it('should have the correct styles', () => {   
     expect(wrapper.find(ScheduleForm).props().styles).toBe(styles);
  });
});