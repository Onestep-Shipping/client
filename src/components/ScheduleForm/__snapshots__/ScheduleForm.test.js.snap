// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScheduleForm should render the ScheduleForm Component correctly 1`] = `[Function]`;
